packages/codec/src/base.ts

```diff
@@ -114,3 +114,90 @@ export function createFixedBytesCodec<Unpacked, Packable = Unpacked>(
     }),
   };
 }
+
+const LAST_ACCESS_PATH = Symbol("PACKABLE_ACCESS_PATH");
+
+export class CodecBaseParseError extends Error {
+  constructor(message: string, public expectedType: string) {
+    super(message);
+  }
+}
+
+class CodecPackableProxyHandler<T extends Record<any, unknown> | any[]>
+  implements ProxyHandler<T>
+{
+  constructor(
+    private previousAccessPaths: PropertyKey[] = [],
+    private onAccess?: (paths: PropertyKey[]) => unknown
+  ) {}
+
+  lastAccess: PropertyKey[] = [];
```

Comment:
so the `lastAccess` is a stateful prop here? if a developer create a custom codec, it is working correctly? like this

```ts
const codec = createCodec({
  pack(obj) {
     const a = obj.a;
     const b = obj.b;

     return concat(bytify(a) /* error here */, bytify(b));
  }
})
```